2.13/mocks/mockOSB.js

```diff
@@ -187,11 +199,31 @@ app.delete('/v2/service_instances/:instance_id', function (req, res) {
 
   if (!req.query.service_id || !req.query.plan_id) {
     res.sendStatus(400)
-  } else {
-    res.status(202).send({
-      'operation': 'task_10'
-    })
+    return
+  }
+  
+  var found = findWhichContains(serviceInstances, 'instance_id', req.params.instance_id)
+  if (!found) {
+    res.status(410).send({})
+    return
+  }
+  var i = serviceInstances.indexOf(found)
+  if (i > -1) {
+    delete serviceInstances[i]
+  }
+
+  // Although the spec doesn't fomulate how to deal with related binding resource when deleting
+  // service instances, the mock server would remove all binding info related to the instance.

```

Comment:
According to Zhongyi's [comment](https://github.com/openservicebrokerapi/osb-checker/pull/69#discussion_r236111194), you may want to stop saying `all binding info` or make below statements in a loop, and I prefer the latter one.

Reply:
maybe following code can help 
```javascript
do {
    found = findWhichContains(serviceBindings, 'instance_id', req.params.instance_id)
    var k = serviceBindings.indexOf(found)
    if (k > -1) {
      delete serviceBindings[k]
    }
  } while (k > -1);
```

Reply:
Got it, thanks!